### PR TITLE
Add full pantheon build actions

### PR DESCRIPTION
**Pull request by srtfisher**
> ⚠️ Pending the full release and versioning of https://github.com/alleyinteractive/action-deploy-to-remote-repository

- Adds GitHub actions for building the project via https://github.com/alleyinteractive/action-deploy-to-remote-repository/pull/1 with full examples that should get a site deployed after filling in the repository URL/ID.
- Adds a `.deployignore` file.

### CODE REVIEW

**Review by renatonascalves — APPROVED** [review]
🌵

File: .deployignore [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Caches
+.npm
+.phpcs.cache.json
+.phpunit.result.cache
+
+# Dependencies
+node_modules
+vendor
```
Comment (kevinfodness) [inline .deployignore:8]: Is this correct? Would we want to install Composer dependencies used in production on Pantheon in a `vendor` folder? I know that for VIP we put this in `client-mu-plugins` but on Pantheon we may want to just use the standard `vendor` folder.

File: .pantheon/pantheon.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+# For more information, see: https://pantheon.io/docs/pantheon-yml/
+api_version: 1
+
+# See https://pantheon.io/docs/pantheon-yml#protected-web-paths for usage.
+protected_web_paths:
+  - /wp-content/.gitignore
+  - /wp-content/.gitmodules
+  - /wp-content/.phpcs.xml
+  - /wp-content/.pantheon
+  - /wp-content/vendor
```
Comment (kevinfodness) [inline .pantheon/pantheon.yml:11]: We're expecting the vendor path to live here, so we shouldn't be deployignoring it.

File: .pantheon/pantheon.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Put overrides to your pantheon.upstream.yml file here.
+# For more information, see: https://pantheon.io/docs/pantheon-yml/
+api_version: 1
+
+# See https://pantheon.io/docs/pantheon-yml#protected-web-paths for usage.
+protected_web_paths:
+  - /wp-content/.gitignore
+  - /wp-content/.gitmodules
+  - /wp-content/.phpcs.xml
```
Comment (kevinfodness) [inline .pantheon/pantheon.yml:9]: Alternately, could we deployignore development files like these so they don't make it to Pantheon's systems at all? We could do the same with test folders, and even with source folders that are transpiled (e.g., include the `build` folder in the plugin, but not the `blocks` or `entries` folder).

File: .pantheon/pantheon.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  - /wp-content/vendor
+  - /wp-content/composer.json
+  - /wp-content/composer.lock
+  - /wp-content/README.md
+
+# See https://pantheon.io/docs/pantheon-yml/#enforce-https--hsts for valid values.
+enforce_https: full
+
+# See https://docs.pantheon.io/pantheon-yml#php-version
+php_version: 8.1
```
Comment (kevinfodness) [inline .pantheon/pantheon.yml:20]: If we're going to use 8.1 here we ought to use it in this project as a whole. Could you set this to 8.0 for now and file an issue to bump the repo to 8.1? We'll need to update composer.json and rebuild the lockfile.

File: ci-templates/.github/workflows/deploy-to-pantheon-live.yml [hunk 5]
```diff
@@ -6,28 +6,63 @@ on:
...
-  build:
+  build-and-sync:
     runs-on: ubuntu-latest
     steps:
-    - name: Perform build steps
-      run: echo "Perform your build steps"
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        submodules: 'recursive'
```
Comment (kevinfodness) [inline ci-templates/.github/workflows/deploy-to-pantheon-live.yml:15]: I'd really love to kill submodules for good. Composer is a better method for installing dependencies.

File: ci-templates/.github/workflows/deploy-to-pantheon-live.yml [hunk 6]
```diff
@@ -6,28 +6,63 @@ on:
...
+    #     key: ${{ runner.os }}-webpack-themes
 
-    secrets:
-      REMOTE_REPO_SSH_KEY: ${{ secrets.ALLEY_CI_PANTHEON_SSH_KEY }}
+    - name: Setup Node
+      uses: actions/setup-node@v3
+      with:
+        cache: 'npm'
+        cache-dependency-path: package-lock.json
+        node-version: 16
```
Comment (kevinfodness) [inline ci-templates/.github/workflows/deploy-to-pantheon-live.yml:29]: This project uses node 18 so we'll need to reflect that everywhere.

File: ci-templates/.github/workflows/deploy-to-pantheon-live.yml [hunk 7]
```diff
@@ -6,28 +6,63 @@ on:
...
 
-  sync-to-pantheon:
-    needs: build
-    uses: alleyinteractive/.github/.github/workflows/deploy-to-remote-repository.yml@main
-    with:
-      remote_repo: 'ssh://codeserver.dev.PANTHEON-SITE-ID@codeserver.dev.PANTHEON-SITE-ID.drush.in:2222/~/repository.git'
-      remote_branch: 'master' # Notable that this differs from 'production'
-      destination_directory: 'wp-content/'
-      exclude_list: '.git, .gitmodules, node_modules'
+    # - name: Cache Theme Webpack Folder
```
Comment (kevinfodness) [inline ci-templates/.github/workflows/deploy-to-pantheon-live.yml:17]: Why is this commented out?

File: ci-templates/.github/workflows/deploy-to-pantheon-live.yml [hunk 8]
```diff
@@ -6,28 +6,63 @@ on:
...
+      uses: actions/setup-node@v3
+      with:
+        cache: 'npm'
+        cache-dependency-path: package-lock.json
+        node-version: 16
+
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: 8.1
```
Comment (kevinfodness) [inline ci-templates/.github/workflows/deploy-to-pantheon-live.yml:34]: Same comment here re: PHP version.

File: ci-templates/.github/workflows/deploy-to-pantheon-live.yml [hunk 9]
```diff
@@ -6,28 +6,63 @@ on:
...
+        composer-options: "--no-progress --no-ansi --no-interaction --prefer-dist --no-dev"
+
+    - name: Install npm dependencies
+      run: npm ci
+
+    - name: Run npm build
+      run: npm run build
+
+    - name: Sync to Pantheon
+      uses: alleyinteractive/action-deploy-to-remote-repository@feature # TODO: Update before merging
```
Comment (kevinfodness) [inline ci-templates/.github/workflows/deploy-to-pantheon-live.yml:51]: This is merged now.

**Review by kevinfodness — CHANGES_REQUESTED** [review]

File: .pantheon/pantheon.yml (hunk 4 above)
Comment (kevinfodness) [reply .pantheon/pantheon.yml:20]: Alternately, we could go directly to 8.2. It's available on Pantheon and recommended.

File: ci-templates/.github/workflows/deploy-to-pantheon-multisite.yml [hunk 10]
```diff
@@ -7,19 +7,53 @@ on:
...
+      run: npm ci
+
+    - name: Run npm build
+      run: npm run build
+
+    - name: Sync to Pantheon
+      uses: alleyinteractive/action-deploy-to-remote-repository@feature
+      with:
+        remote_repo: 'ssh://user@host:2222/~/repository.git'
+        remote_branch: 'staging'
```
Comment (kevinfodness) [inline ci-templates/.github/workflows/deploy-to-pantheon-multisite.yml:55]: We should make this dynamic based on the branch name that triggered the workflow. Recommendation:
```suggestion
        remote_branch: ${{ github.ref_name }}
```

**Comment by kevinfodness** [comment]
See also #30

**Comment by kevinfodness** [comment]
Rolled up into #30 - further modifications can be made in a followup feature branch / PR as necessary

**Pull request closed without merging**